tsfile/src/main/java/org/apache/iotdb/tsfile/encoding/encoder/TSEncodingBuilder.java

```diff
@@ -105,24 +101,51 @@ public Encoder getEncoder(TSDataType type) {
     public void initFromProps(Map<String, String> props) {
       // set max error from initialized map or default value if not set
       if (props == null || !props.containsKey(Encoder.MAX_STRING_LENGTH)) {
-        maxStringLength = conf.maxStringLength;
+        maxStringLength = TSFileConfig.maxStringLength;
       } else {
         maxStringLength = Integer.valueOf(props.get(Encoder.MAX_STRING_LENGTH));
         if (maxStringLength < 0) {
-          maxStringLength = conf.maxStringLength;
+          maxStringLength = TSFileConfig.maxStringLength;
           LOG.warn("cannot set max string length to negative value, replaced with default value:{}",
               maxStringLength);
         }
       }
     }
   }
 
+  private static abstract class rleAndTs2diffCommon extends TSEncodingBuilder {
+
+    protected int maxPointNumber;
+
+    @Override
+    /**
+     * Both RLE and TS_2DIFF could specify <b>max_point_number</b> in given JSON Object, which means the maximum
+     * decimal digits for float or double data.
+     */
+    public void initFromProps(Map<String, String> props) {
+      // set max error from initialized map or default value if not set
+      if (props == null || !props.containsKey(Encoder.MAX_POINT_NUMBER)) {
+        maxPointNumber = TSFileConfig.floatPrecision;
+      } else {
+        maxPointNumber = Integer.valueOf(props.get(Encoder.MAX_POINT_NUMBER));
+        if (maxPointNumber < 0) {
+          maxPointNumber = TSFileConfig.floatPrecision;
+          LOG.warn("cannot set max point number to negative value, replaced with default value:{}",
+              maxPointNumber);
+        }
+      }
+    }
+
+  }
+
   /**
    * for ENUMS, INT32, BOOLEAN, INT64, FLOAT, DOUBLE.
    */
-  public static class RLE extends TSEncodingBuilder {
+  public static class RLE extends rleAndTs2diffCommon {
```

Comment:
The name `rleAndTs2diffCommon` is not good.

Reply:
These kinds of modifications will be revoked.